src/usb_moded-network.c

```diff
@@ -299,18 +300,16 @@ static int resolv_conf_dns(struct ipforward_data *ipforward)
 
 static int checklink(void)
 {
-   char *dest = malloc( 32 * sizeof(char)) ;
-   int ret = 0;
-
-   readlink("/etc/udhcpd.conf", dest, 32);
-   if(dest != NULL)
-   {
-	strcpy(&dest[31], "\0");
-	ret = (strcmp(dest, UDHCP_CONFIG_PATH));
-   }
-   free(dest);
-
-   return(ret);
+  int ret = -1;
+  const size_t bufsiz = 32;
+  char *dest = malloc(bufsiz);
```

Comment:
This heap allocation is there for the same reason why it was there in the original code. I don't know why it was there.

If I wrote the original code, I would either allocate a small amount (such as 32 bytes) on stack or a large amount (such as PATH_MAX) from the heap, to reduce the chance of running out of stack space.

When I'm fixing a problem, I am trying to be consistent with the existing code.